src/containers/forms/CreateCategoryForm.jsx

```diff
@@ -0,0 +1,189 @@
+      return {
+        value: '',
+        name: '',
+        description: '',
+        errors: [],
+        message: nextProps.message,
+      };
+    }
+    return {
+      message: nextProps.message,
+    };
+  };
+  /**
+   * @memberOf CreateCategoryForm
+   * change event handler
+   * @param {Event} event - change event on select input
+   */
+  handleChange = (event) => {
+    this.setState({ [event.target.name]: event.target.value });
+    // if input value is not empty remove it from error list
+    if (!this.state[event.target.name]) {
+      const errors = this.state.errors.filter(error => !error.includes(event.target.name));
+      this.setState({ errors });
+    }
+  }
+
+  handleAddEvent = (event) => {
+    event.preventDefault();
+    const { name, description, value } = this.state;
+    const category = {
+      name,
+      description,
+      value,
+    };
+    this.setState({
+      errors: validateFormFields(category),
+    }, () => {
+      if (this.state.errors.length === 0) {
+        this.props.createCategory(category);
+      }
+    });
+  }
+
+  /**
+   * @name resetState
+   * @summary resets state to clear form fields and error messages
+   */
+  resetState = () => {
+    this.setState({
+      name: '',
+      value: '',
+      description: '',
+      errors: [],
+      message: null,
+    });
+  }
+
+  /**
+   * @name cancelModal
+   * @summary reset state and close modal
+   */
+  cancelModal = (event) => {
+    event.preventDefault();
+    this.resetState();
+    this.props.closeModal();
+  }
+
+  renderValidationError = (field, replaceWord) => {
+    if (this.state.errors.indexOf(field) >= 0) {
+      return `${capitalizeString(replaceWord || field)} is required`;
+    }
+    return '';
+  }
+
+  render() {
+    const { message } = this.state;
+
+    return (
+      <form>
+        <div className='titleForm'>Create a Category</div>
+        <SingleInput
+          type='text'
+          name='name'
+          title='Name'
+          value={this.state.name}
+          handleChange={this.handleChange}
+        />
+        <span className='validate__errors'>
+          {this.renderValidationError('name')}
+        </span>
+        <SingleInput
+          type='number'
+          name='value'
+          value={this.state.value}
+          title='Number of points'
+          handleChange={this.handleChange}
+        />
+        <span className='validate__errors'>
+          {this.renderValidationError('value')}
+        </span>
+        <FormError errors={this.state.errors} fieldName='points' />
+        <TextArea
+          title='Description'
+          rows={5}
+          resize={false}
+          name='description'
+          placeholder='keep it brief'
+          handleChange={this.handleChange}
+          value={this.state.description}
+        />
+        <span className='validate__errors'>
+          {this.renderValidationError('description')}
```

Comment:
Avoid calling functions within the return for the JSX but instead evaluate before the return and pass a constant here. Just for best practice.